app/Http/Respositories/Eloquent/InvoiceRepository.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Http\Respositories\Eloquent;
+
+use App\Http\Respositories\InvoiceRepositoryInterface;
+use App\Models\Invoice;
+
+class InvoiceRepository extends BaseRepository implements InvoiceRepositoryInterface
+{
+    protected $model;
+
+    /**
+     * @param $model
+     */
+    public function __construct(Invoice $model)
+    {
+        parent::__construct($model);
+        $this->model = $model;
+    }
+
+    public function deleteWithRelationship($request)
+    {
+        $invoice = $this->model->with('invoiceproduct')->where('id', $request->input('id'))->first();
+    }
+
+    public function getConditionsWithRelationship($invoiceProduct)
+    {
+        return $this->model->with('invoiceproduct')->where('id', $invoiceProduct)->get();
+    }
+
+    public function filterSearchRepository($request)
+    {
+        $selectionget = $request->selection;
```

Comment:
camelCase
And I think you don't need this selection. You must use orWhere to get data when 1 in 3 below conditions matching